### PR TITLE
fix(form): match experimental spec

### LINKED ISSUE #1966: [Audit] Form
Stub

### PR DESCRIPTION
**Pull request by emyarod**
Closes #1966

This PR incorporates feedback from the March 2019 form component audit

Related: https://github.com/IBM/carbon-components/pull/1965

### CODE REVIEW

File: src/components/form/_form.scss [hunk 1]
```diff
@@ -182,7 +182,7 @@
   }
 
   .#{$prefix}--form__helper-text {
-    @include typescale('omega');
+    @include type-style('helper-text-01');
     font-style: italic;
     color: $text-02;
     line-height: 1.5;
```
Comment (aagonzales) [inline src/components/form/_form.scss]: The height is still wrong should be `line-height: 1` (or 16px). Its actually built into the type style: 
image

**Review by aagonzales — COMMENTED** [review]
None of the form elements had helper text so I could only confirm the changes were made via the code.

File: src/components/form/_form.scss (hunk 1 above)
Comment (emyarod) [reply src/components/form/_form.scss]: thanks for catching that, I updated the style rules and added helper text to the examples in the experimental docs

**Review by tw15egan — APPROVED** [review]
👍

**Review by aagonzales — APPROVED** [review]
Perfect!

File: src/components/form/_form.scss [hunk 2]
```diff
@@ -114,6 +114,7 @@
 
   .#{$prefix}--form-item {
     @include type-style('body-short-01');
+    @include carbon--font-family('sans');
```
Comment (asudoh) [inline src/components/form/_form.scss]: I thought `type-style('body-short-01')` contains `carbon--font-family('sans')` - Did you find otherwise?

**Review by asudoh — COMMENTED** [review]
Just one comment to the code - BTW good if you can quickly see if `v9` looks OK, given I see markup changes to non-roped-off code. Thanks @emyarod!

File: src/components/form/_form.scss (hunk 2 above)
Comment (emyarod) [reply src/components/form/_form.scss]: yes I believe that line was added before https://github.com/IBM/carbon-components/pull/1965 was merged and I've now taken it out

Comment (emyarod) [reply src/components/form/_form.scss]: yes I believe that line was added before https://github.com/IBM/carbon-components/pull/1965 was merged and I've now taken it out

**Comment by emyarod** [comment]
@asudoh is there a specific change that you see as not being roped off? Most of the diff is just formatting, and the only significant changes were made behind the `componentsX` flag

**Comment by asudoh** [comment]
L27 and L31-L43 in old version got me such impression, but I may be wrong. Any ideas on where they went?

**Comment by emyarod** [comment]
got it @asudoh, I replaced the `<select>` with a dropdown by mistake but reverted it now

File: src/components/form/form.hbs [hunk 3]
```diff
@@ -7,50 +7,85 @@
...
-    <select id="select-id" class="{{@root.prefix}}--select-input">
-      {{#each selectItems}}
-        {{#if items}}
-          <optgroup class="{{@root.prefix}}--select-optgroup" label="{{label}}">
-            {{#each items}}
-              <option class="{{@root.prefix}}--select-option" value="{{value}}"{{#if disabled}} disabled{{/if}}{{#if selected}} selected{{/if}}{{#if hidden}} hidden{{/if}}>{{label}}</option>
-            {{/each}}
-          </optgroup>
-        {{else}}
-          <option class="{{@root.prefix}}--select-option" value="{{value}}"{{#if disabled}} disabled{{/if}}{{#if selected}} selected{{/if}}{{#if hidden}} hidden{{/if}}>{{label}}</option>
```
Comment (asudoh) [inline src/components/form/form.hbs:40]: Has `<select>` without `<optgroup>` been gone...? Though I see a bug in `master` where `select.config.js` not sending data for `<optgorup>` I see `{{#if}}` as well as `{{else}}` sections here has a solid meaning.

Comment (emyarod) [reply src/components/form/form.hbs:40]: I just updated the markup for the select component to what we have in `master`

Comment (asudoh) [reply src/components/form/form.hbs:40]: Thank you for the update @emyarod!

**Pull request merged**